### DISCUSSION TITLE
Task List Thoughts

### DISCUSSION THREAD

**Discussion by trevor-clarke** [community/community#5173] [Projects and Issues]
Hey everyone! 

I was hoping to suggest adding a new state to the checkbox to add some more flexibility to its use. 

For example: 
![image](https://user-images.githubusercontent.com/13138669/129926811-4029b0c1-8d7c-4a87-824a-35aa7b032ede.png)

In my case, I am working on a project with a bit of a to-do list, and I would like to be able to model 'in progress' using these checkboxes. 

I was thinking something like adding a `-` between the `[  ]` could be how this is reflected.

**Comment by mattleibow** [comment]
This sounds like a good idea! We sometimes have a big issues that links a bunch of smaller tasks together and this will be wonderful!

**Comment by cmadland** [comment]
Seems to me this would be great for tasks with sub-tasks. If the list of sub-tasks is incomplete, then the parent task can have the `-` as 'partially complete'.

**Reply by moorscode** [reply]
When working with multiple people on tasks of a list, having a "claimed" or "in progress" state would be very helpful.

This suggestion would help to achieve that state :)

**Comment by philippspindler** [comment]
We also have the case when working with a pull request template with predefined tasks that sometimes it does not make sense to tick the box because the predefined task is not relevant for this PR. So instead of ticking it, it would be good to mark it as "not relevant" instead of just leaving the task open.

**Reply by benmvp** [reply]
> We also have the case when working with a pull request template with predefined tasks that sometimes it does not make sense to tick the box because the predefined task is not relevant for this PR. So instead of ticking it, it would be good to mark it as "not relevant" instead of just leaving the task open.

I usually use strikethrough for this case:

```md
[ ] ~~Do the thing that sometimes doesn't need to be done every time~~
```

**Reply by philippspindler** [reply]
Good idea! But with this approach you still have to switch to edit mode. That is what I would like to avoid.

**Reply by Zadjil-AR** [reply]
There is a good suggestion in #5248 which would allow the strikethrough without going into edit mode.

**Reply by MaurizioVacca** [reply]
We generally get rid of irrelevant predefined tasks from the template if they're not relevant for the PR. We generally do this whenever we open a new PR, so no further edit is required.

**Comment by sebastian-philipp** [comment]
TBH, this sounds overly complicated to me. I'd rather want to avoid this additional click for the large majority of check boxes where this is not required.

**Comment by MaurizioVacca** [comment]
Imho, the task list is the wrong tool to accomplish this. By definition, checkboxes are binary elements so the state could either be done/not done.

This means that the task list (or check list) should be used just to keep in mind what needs to be done. Task status and progression can be eventually tracked using:

- separated PRs;
- a dedicated task management software.

Best!
